File: sdkwork-rtc-sdk-swift/Sources/RtcSdk/RtcProviderActivationCatalog.swift
public struct RtcProviderActivationCatalogEntry {
    public let providerKey: String
    public let pluginId: String
    public let driverId: String
    public let activationStatus: String
    public let runtimeBridge: Bool
    public let rootPublic: Bool
    public let packageBoundary: Bool
    public let builtin: Bool
    public let packageIdentity: String
}

public enum RtcProviderActivationCatalog {
    public static let recognizedActivationStatuses: [String] = [
        "root-public-builtin",
        "package-boundary",
        "control-metadata-only",
    ]

    public static let entries: [RtcProviderActivationCatalogEntry] = [
        .init(providerKey: "volcengine", pluginId: "rtc-volcengine", driverId: "sdkwork-rtc-driver-volcengine", activationStatus: "control-metadata-only", runtimeBridge: false, rootPublic: false, packageBoundary: false, builtin: true, packageIdentity: "RtcSdkProviderVolcengine"),
        .init(providerKey: "aliyun", pluginId: "rtc-aliyun", driverId: "sdkwork-rtc-driver-aliyun", activationStatus: "control-metadata-only", runtimeBridge: false, rootPublic: false, packageBoundary: false, builtin: true, packageIdentity: "RtcSdkProviderAliyun"),
        .init(providerKey: "tencent", pluginId: "rtc-tencent", driverId: "sdkwork-rtc-driver-tencent", activationStatus: "control-metadata-only", runtimeBridge: false, rootPublic: false, packageBoundary: false, builtin: true, packageIdentity: "RtcSdkProviderTencent"),
        .init(providerKey: "agora", pluginId: "rtc-agora", driverId: "sdkwork-rtc-driver-agora", activationStatus: "control-metadata-only", runtimeBridge: false, rootPublic: false, packageBoundary: false, builtin: false, packageIdentity: "RtcSdkProviderAgora"),
        .init(providerKey: "zego", pluginId: "rtc-zego", driverId: "sdkwork-rtc-driver-zego", activationStatus: "control-metadata-only", runtimeBridge: false, rootPublic: false, packageBoundary: false, builtin: false, packageIdentity: "RtcSdkProviderZego"),
        .init(providerKey: "livekit", pluginId: "rtc-livekit", driverId: "sdkwork-rtc-driver-livekit", activationStatus: "control-metadata-only", runtimeBridge: false, rootPublic: false, packageBoundary: false, builtin: false, packageIdentity: "RtcSdkProviderLivekit"),
        .init(providerKey: "twilio", pluginId: "rtc-twilio", driverId: "sdkwork-rtc-driver-twilio", activationStatus: "control-metadata-only", runtimeBridge: false, rootPublic: false, packageBoundary: false, builtin: false, packageIdentity: "RtcSdkProviderTwilio"),
        .init(providerKey: "jitsi", pluginId: "rtc-jitsi", driverId: "sdkwork-rtc-driver-jitsi", activationStatus: "control-metadata-only", runtimeBridge: false, rootPublic: false, packageBoundary: false, builtin: false, packageIdentity: "RtcSdkProviderJitsi"),
        .init(providerKey: "janus", pluginId: "rtc-janus", driverId: "sdkwork-rtc-driver-janus", activationStatus: "control-metadata-only", runtimeBridge: false, rootPublic: false, packageBoundary: false, builtin: false, packageIdentity: "RtcSdkProviderJanus"),
        .init(providerKey: "mediasoup", pluginId: "rtc-mediasoup", driverId: "sdkwork-rtc-driver-mediasoup", activationStatus: "control-metadata-only", runtimeBridge: false, rootPublic: false, packageBoundary: false, builtin: false, packageIdentity: "RtcSdkProviderMediasoup"),
    ]

public static func getRtcProviderActivationByProviderKey(_ providerKey: String) -> RtcProviderActivationCatalogEntry? {
        entries.first { $0.providerKey == providerKey }
    }

}

File: sdkwork-rtc-sdk-flutter/lib/src/rtc_provider_activation_catalog.dart
final class RtcProviderActivationCatalogEntry {
  const RtcProviderActivationCatalogEntry({
    required this.providerKey,
    required this.pluginId,
    required this.driverId,
    required this.activationStatus,
    required this.runtimeBridge,
    required this.rootPublic,
    required this.packageBoundary,
    required this.builtin,
    required this.packageIdentity,
  });

  final String providerKey;
  final String pluginId;
  final String driverId;
  final String activationStatus;
  final bool runtimeBridge;
  final bool rootPublic;
  final bool packageBoundary;
  final bool builtin;
  final String packageIdentity;
}

final class RtcProviderActivationCatalog {
  static const List<String> recognizedActivationStatuses = <String>[
    "root-public-builtin",
    "package-boundary",
    "control-metadata-only",
  ];

  static const List<RtcProviderActivationCatalogEntry> entries =
      <RtcProviderActivationCatalogEntry>[
    RtcProviderActivationCatalogEntry(
      providerKey: "volcengine",
      pluginId: "rtc-volcengine",
      driverId: "sdkwork-rtc-driver-volcengine",
      activationStatus: "control-metadata-only",
      runtimeBridge: false,
      rootPublic: false,
      packageBoundary: false,
      builtin: true,
      packageIdentity: "rtc_sdk_provider_volcengine",
    ),
    RtcProviderActivationCatalogEntry(
      providerKey: "aliyun",
      pluginId: "rtc-aliyun",
      driverId: "sdkwork-rtc-driver-aliyun",
      activationStatus: "control-metadata-only",
      runtimeBridge: false,
      rootPublic: false,
      packageBoundary: false,
      builtin: true,
      packageIdentity: "rtc_sdk_provider_aliyun",
    ),
    RtcProviderActivationCatalogEntry(
      providerKey: "tencent",
      pluginId: "rtc-tencent",
      driverId: "sdkwork-rtc-driver-tencent",
      activationStatus: "control-metadata-only",
      runtimeBridge: false,
      rootPublic: false,
      packageBoundary: false,
      builtin: true,
      packageIdentity: "rtc_sdk_provider_tencent",
    ),
    RtcProviderActivationCatalogEntry(
      providerKey: "agora",
      pluginId: "rtc-agora",
      driverId: "sdkwork-rtc-driver-agora",
      activationStatus: "control-metadata-only",
      runtimeBridge: false,
      rootPublic: false,
      packageBoundary: false,
      builtin: false,
      packageIdentity: "rtc_sdk_provider_agora",
    ),
    RtcProviderActivationCatalogEntry(
      providerKey: "zego",
      pluginId: "rtc-zego",
      driverId: "sdkwork-rtc-driver-zego",
      activationStatus: "control-metadata-only",
      runtimeBridge: false,
      rootPublic: false,
      packageBoundary: false,
      builtin: false,
      packageIdentity: "rtc_sdk_provider_zego",
    ),
    RtcProviderActivationCatalogEntry(
      providerKey: "livekit",
      pluginId: "rtc-livekit",
      driverId: "sdkwork-rtc-driver-livekit",
      activationStatus: "control-metadata-only",
      runtimeBridge: false,
      rootPublic: false,
      packageBoundary: false,
      builtin: false,
      packageIdentity: "rtc_sdk_provider_livekit",
    ),
    RtcProviderActivationCatalogEntry(
      providerKey: "twilio",
      pluginId: "rtc-twilio",
      driverId: "sdkwork-rtc-driver-twilio",
      activationStatus: "control-metadata-only",
      runtimeBridge: false,
      rootPublic: false,
      packageBoundary: false,
      builtin: false,
      packageIdentity: "rtc_sdk_provider_twilio",
    ),
    RtcProviderActivationCatalogEntry(
      providerKey: "jitsi",
      pluginId: "rtc-jitsi",
      driverId: "sdkwork-rtc-driver-jitsi",
      activationStatus: "control-metadata-only",
      runtimeBridge: false,
      rootPublic: false,
      packageBoundary: false,
      builtin: false,
      packageIdentity: "rtc_sdk_provider_jitsi",
    ),
    RtcProviderActivationCatalogEntry(
      providerKey: "janus",
      pluginId: "rtc-janus",
      driverId: "sdkwork-rtc-driver-janus",
      activationStatus: "control-metadata-only",
      runtimeBridge: false,
      rootPublic: false,
      packageBoundary: false,
      builtin: false,
      packageIdentity: "rtc_sdk_provider_janus",
    ),
    RtcProviderActivationCatalogEntry(
      providerKey: "mediasoup",
      pluginId: "rtc-mediasoup",
      driverId: "sdkwork-rtc-driver-mediasoup",
      activationStatus: "control-metadata-only",
      runtimeBridge: false,
      rootPublic: false,
      packageBoundary: false,
      builtin: false,
      packageIdentity: "rtc_sdk_provider_mediasoup",
    ),
      ];

  const RtcProviderActivationCatalog._();
}

RtcProviderActivationCatalogEntry? getRtcProviderActivationByProviderKey(String providerKey) {
  for (final entry in RtcProviderActivationCatalog.entries) {
    if (entry.providerKey == providerKey) {
      return entry;
    }
  }

  return null;
}
